src/shared/bootspec.c

```diff
@@ -591,8 +591,7 @@ static int boot_entries_select_default(const BootConfig *config) {
         assert(config->entries || config->n_entries == 0);
 
         if (config->n_entries == 0) {
-                log_debug("Found no default boot entry :(");
-                return -1; /* -1 means "no default" */
+                return log_debug_errno(-1, "Found no default boot entry :("); /* -1 means "no default" */
```

Comment:
this should remain a regular "return -1". -1 is not really an error here, just a special marker. if we pass it to log_debug_errno() it would be added as ERRNO= field to structured logging if structured logging is used. And we really don't want that.